File: app/src/main/java/org/cerion/stockcharts/model/Symbol.java
package org.cerion.stockcharts.model;

public class Symbol {

    private String symbol;
    private String name;
    private String exchange;

    public Symbol(String symbol) {
        this.symbol = symbol;
    }

    public Symbol(String symbol, String name, String exchange) {
        this.symbol = symbol;
        this.name = name;
        this.exchange = exchange;
    }

    public String getSymbol() {
        return symbol;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getExchange() {
        return exchange;
    }

    public void setExchange(String exchange) {
        this.exchange = exchange;
    }
}

File: app/src/main/java/org/cerion/stockcharts/database/Symbol.kt
package org.cerion.stockcharts.database

import androidx.room.*
import org.cerion.marketdata.core.model.Symbol


@Entity(tableName = "symbols")
data class SymbolEntity(
        @PrimaryKey
        val symbol: String,
        val name: String,
        val exchange: String)

@Dao
interface SymbolDao {
    @Query("select * from symbols")
    fun getAll(): List<SymbolEntity>

    @Insert
    fun insert(symbol: SymbolEntity)

    @Delete
    fun delete(symbol: SymbolEntity)

}

fun SymbolEntity.toSymbol(): Symbol = Symbol(symbol, name, exchange)